### PR TITLE
ARROW-2344: [Go] Run Go unit tests in Travis CI

### PR DESCRIPTION
**Pull request by sbinet**

### CODE REVIEW

**Comment by wesm** [comment]
Thanks @sbinet! We should also integrate this into https://github.com/apache/arrow/blob/master/ci/detect-changes.py so the Go tests are only run when there is a Go change.

**Comment by sbinet** [comment]
should I integrate this into this PR or should I create a JIRA ticket+new PR ?

**Comment by wesm** [comment]
You can do it here, no problem

**Comment by sbinet** [comment]
Ok. PTAL.

**Comment by sbinet** [comment]
not sure why `appveyor` failed (in C++, I think) nor why suddenly the `travis-ruby` test failed (loading some lua library)...

**Comment by sbinet** [comment]
(but this seems unrelated to my change.)

File: .travis.yml [hunk 1]
```diff
@@ -217,6 +217,15 @@ matrix:
     - mkdir -p target/kcov
     - RUST_BACKTRACE=1 cargo coverage --verbose
     - bash <(curl -s https://codecov.io/bash) || echo "Codecov did not collect coverage reports"
+  # Go
+  - language: go
+    os: linux
+    go:
+    - 1.10.x
+    script:
+    - go get -d -t -v ./go/arrow/...
```
Comment (xhochy) [inline .travis.yml]: Can you add a line `- if [ $ARROW_CI_GO_AFFECTED != "1" ]; then exit; fi` here so that the Go build only runs for GO PRs?

**Review by xhochy — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (sbinet) [reply .travis.yml]: Sure. Will do when in front of a real keyboard.

Comment (wesm) [reply .travis.yml]: Rebased and added that

**Comment by pitrou** [comment]
@sbinet yes, the `test_use_huge_pages` failure in unrelated.

**Review by pitrou — APPROVED** [review]
+1. Thanks for doing this @sbinet !

**Comment by pitrou** [comment]
Just for my personal enlignment: when `go install` runs in 0.10s, is it really the entire build time for Arrow? :-o

**Comment by sbinet** [comment]
Yep. Go builds really fast. :)
(Also, Go arrow is rather bare bones for now...)

**Comment by wesm** [comment]
The way I look at Go, it's Google's reaction to a decade of production C++ development. A lot of Google C++ best practices are reflected in the language itself, and the difference in compiler speed of course is a stark difference to large C++ projects

**Pull request closed without merging**